### PR TITLE
Basic UI for adding biblio fields

### PR DESCRIPTION
**Pull request by RoySegall**
#99

a lot
list

### CODE REVIEW

File: modules/biblio_ui/biblio_ui.module [hunk 1]
```diff
@@ -687,6 +704,88 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+
+/**
+ * Import a single predefined biblio type.
+ */
+function biblio_ui_import_single_biblio_type($type) {
+  $info = biblio_get_types_info($type);
+  biblio_add_publication_types(array($type));
+  drupal_set_message(t('The %title has creates successfully.', array('%title' => $info['name'])));
+  drupal_goto('admin/structure/biblio/import');
+  return '';
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: Probably not needed.

File: modules/biblio_ui/biblio_ui.module [hunk 2]
```diff
@@ -687,6 +704,88 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+  batch_process('admin/structure/biblio/import');
+}
+
+/**
+ * Import a single predefined biblio type.
+ */
+function biblio_ui_import_single_biblio_type($type) {
+  $info = biblio_get_types_info($type);
+  biblio_add_publication_types(array($type));
+  drupal_set_message(t('The %title has creates successfully.', array('%title' => $info['name'])));
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: `The %title was created successfully.`

**Comment by RoySegall** [comment]
add link
selected field
added field

File: modules/biblio_ui/biblio_ui.module [hunk 3]
```diff
@@ -687,6 +713,231 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+/**
+ * Manage the fields of the biblio bundle.
+ */
+function biblio_ui_create_fields($form, &$form_state, $bundle) {
+  // Save the bundle for later.
+  $form_state['bundle'] = $bundle;
+
+  // Get the existed fields and the fields which not yet added.
+  $fields = biblio_fields_info();
+  $existed_fields = field_info_instances('biblio', $bundle);
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: existed => existing

File: modules/biblio_ui/biblio_ui.module [hunk 4]
```diff
@@ -105,6 +105,23 @@ function biblio_ui_menu() {
     'type' => MENU_LOCAL_ACTION,
   );
 
+  $items['admin/structure/biblio/import'] = array(
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: `attach-fields`

**Comment by RoySegall** [comment]
add
alot

File: biblio.module [hunk 5]
```diff
@@ -658,20 +658,34 @@ function biblio_fields_info($field_name = NULL) {
...
     $row->name = $name;
-    $row->description = $values['description'];
+    $row->description = $info['description'];
+
     biblio_type_save($row);
   }
-}
 
+  // Clearing the menu cache after creating a new menu biblio type.
+  drupal_flush_all_caches();
```
Comment (amitaibu) [inline biblio.module]: `variable_set('menu_needs_rebuild', TRUE);`

File: modules/biblio_ui/biblio_ui.module [hunk 6]
```diff
@@ -687,6 +705,166 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+  return $form;
+}
+
+/**
+ * Submit handler; Attach biblio fields to selected biblio types.
+ */
+function biblio_ui_attach_biblio_fields_submit($form, $form_state) {
+  // Import all the selected biblio types via batch api.
+  $operations = array();
+  foreach ($form_state['values']['biblio_types'] as $enable) {
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: looks weird.

File: modules/biblio_ui/biblio_ui.module [hunk 7]
```diff
@@ -687,6 +705,166 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+    if (!$enable) {
+      continue;
+    }
+
+    // Adding the fields for the bundle.
+    $operations[] = array('biblio_create_fields_by_bundle', array(array($enable)));
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: another line.

File: modules/biblio_ui/biblio_ui.module [hunk 8]
```diff
@@ -687,6 +705,166 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+    '#prefix' => '<div id="field-info-wrapper">',
+    '#suffix' => '</div>',
+    '#parents' => array('field_info_wrapper'),
+  );
+
+  $form['field_info_wrapper']['field_name'] = array(
+    '#type' => 'select',
+    '#title' => t('Select field to add'),
+    '#description' => t('Select the field you would like to add.'),
+    '#options' => array('---' => t('Select a field')) + $options,
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: `---` => `''`

File: modules/biblio_ui/biblio_ui.module [hunk 9]
```diff
@@ -687,6 +705,166 @@ function biblio_ui_delete_bundle_submit($form, $form_state) {
...
+
+/**
+ * Submit handler; Attaching the field on the bundle.
+ */
+function biblio_ui_create_fields_submit($form, $form_state) {
+  biblio_create_field($form_state['values']['field_name'], 'biblio', $form_state['bundle']);
+
+  $field_info = biblio_fields_info($form_state['values']['field_name']);
+  $type = biblio_types($form_state['bundle']);
+  drupal_set_message(t('The field %name added to the bundle %bundle succesfully.', array(
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: `$params`

File: biblio.module [hunk 10]
```diff
@@ -658,20 +658,34 @@ function biblio_fields_info($field_name = NULL) {
...
+ * @param Array $types
+ *  List of specific types to create.
+ *
  * @todo: Make the bundles configurable?
  */
-function biblio_add_publication_types() {
-  foreach (biblio_get_types_info() as $values) {
-    $name = $values['name'];
+function biblio_add_publication_types($types = array()) {
+  $default_types = biblio_get_types_info();
```
Comment (amitaibu) [inline biblio.module]: not just default, all declared.

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**